src/EditorFeatures/Core.Wpf/Suggestions/AsyncSuggestedActionsSource.cs

```diff
@@ -111,41 +111,123 @@ private partial class AsyncSuggestedActionsSource : SuggestedActionsSource, IAsy
                     // items should be pushed higher up, and less important items shouldn't take up that much space.
                     var currentActionCount = 0;
 
-                    using var _2 = ArrayBuilder<SuggestedActionSet>.GetInstance(out var lowPrioritySets);
+                    // Builders to store low and medium priority sets that were reported for a higher request priority.
+                    // These mismatched priority sets are stored in one of the below pending priority sets, and later added
+                    // when we are computing sets with matching request priority.
+                    // Note that lowest request priority is reserved for suppression/configuration action sets,
+                    // and we expect the code fix service to only return suppression/configuration actions for this
+                    // request priority. We have an assert for this invariant below and don't need to track pendingLowestPrioritySets.
```

Comment:
thanks for the assert!

Reply:
I am going to revert all the changes in this file. Thanks for your PR to clean up this stuff.